### PR TITLE
gssproxy: check result of calloc() is necessary

### PR DESCRIPTION
**Pull request by GYShit**
Signed-off-by: gaoyusong <gaoyusong2@huawei.com>

### CODE REVIEW

File: src/gssproxy.c [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include "gp_proxy.h"
 #include <signal.h>
 #include <string.h>
+#include <errno.h>
```
Comment (simo5) [inline src/gssproxy.c:9]: Please drop this

File: src/gssproxy.c [hunk 2]
```diff
@@ -114,6 +115,11 @@ int main(int argc, const char *argv[])
     }
 
     gpctx = calloc(1, sizeof(struct gssproxy_ctx));
+    if (!gpctx) {
+        ret = ENOMEM;
```
Comment (simo5) [inline src/gssproxy.c:119]: Please change this to EXIT_FAILURE like for all other error sites in the main() function.

**Review by simo5 — CHANGES_REQUESTED** [review]

File: src/gssproxy.c (hunk 1 above)
Comment (GYShit) [reply src/gssproxy.c]: done

File: src/gssproxy.c (hunk 2 above)
Comment (GYShit) [reply src/gssproxy.c]: done

**Comment by simo5** [comment]
Thank you

**Pull request merged**